docs/topics/cache.txt

```diff
@@ -169,6 +169,121 @@ particularly temporary.
+    }
+
+If you have multiple Redis servers set up in replication mode, you can specify
+the servers either as a semicolon or comma delimited string, or as a list.
+While using multiple servers, all the writes are performed on the lead server.
+Read operations are performed on one of the replicas chosen at random. The first
+server address must be the leader server and the other server addresses must be
+the followers (replicas of leader)::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': [
+                'redis://127.0.0.1:6379', # leader
+                'redis://127.0.0.1:6378', # read-replica 1
+                'redis://127.0.0.1:6377', # read-replica 2
+            ],
+        }
+    }
+
+Along with the standard :ref:`cache_arguments`, the Redis caching backend also
+accepts the following additional options:
+
+* ``db``: By default, Redis ships with 16 logical databases. A database of your
+  choice can be selected either by setting it in the ``LOCATION`` as querystring
+  or path argument, or by setting it in the ``OPTIONS``. The default choice for
+  the database is 0. We can set the database in the following manner::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'redis://127.0.0.1:6379',
+            'OPTIONS': {
+                'db': '10'
+            }
+        }
+    }
+
+  Where you have multiple Redis instances the same database must be selected 
+  for each one. The ``OPTIONS`` ``db`` argument can be used in such cases for 
+  clarity.
+
+* ``pool_class``: Connection pools are a set of connections which are used as 
+  required. ``pool_class`` allows you to set a custom connection pool class.
+  Defaults to ``redis.ConnectionPool``. See the ``redis-py`` `connection pool
+  docs`_ for more details .
+  
+  .. _`connection pool docs`: https://github.com/andymccurdy/redis-py#connection-pools
+
+* ``parser_class``: Parser classes control on how the responses from the
+  Redis server are parsed. By default ``hiredis`` will be used if installed
+  otherwise the ``PythonParser`` will be used. ``Hiredis`` is written in C and
+  therefore can give up to 10x gains over the ``PythonParser``.
+  See the ``redis-py`` docs for more about `parsers`_.
+
+  .. _`parsers`: https://github.com/andymccurdy/redis-py#parsers
+
+In the following example, we set the ``parser_class`` to
+``redis.connection.HiredisParser``, the ``db`` to 12 and the ``pool_class`` to
+``redis.BlockingConnectionPool``::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'unix:///var/run/redis/redis.sock',
+            'OPTIONS': {
+                'pool_class': 'redis.BlockingConnectionPool',
+                'parser_class': 'redis.connection.HiredisParser',
+                'db': 12,
+            }
+        }
+    }
```

Comment:
Extra options for other backends are described shortly in the [_"Cache arguments"_](https://docs.djangoproject.com/en/dev/topics/cache/#cache-arguments) section. IMO we don't need a detailed description and many examples. Maybe:
```
Here's an example configuration for a ``redis`` based backend that chooses the
10th database (by default Redis ships with 16 logical databases), sets a custom
`connection pool class`_ (``redis.ConnectionPool`` is used by default), and ...

    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
            'LOCATION': 'redis://127.0.0.1:6379',
            'OPTIONS': {
                'pool_class': 'redis.BlockingConnectionPool',
                'parser_class': 'redis.connection.HiredisParser',
                'db': '10',
            }
        }   
    }   
    
.. _`connection pool docs`: https://github.com/andymccurdy/redis-py#connection-pools
.. _`parsers`: https://github.com/andymccurdy/redis-py#parsers
```

Reply:
I think a little context with some examples will make things easier for the user. Moreover in the future, we'll be added more options for `username` and `password` too. Also, since there are warnings (in a multi-server setup) and various ways of setting `db` etc, having a separate section would give better organization and clear understanding of the arguments in my opinion.

Reply:
However in the long run, if we write reference documentation, these can move away from here and we can keep a single example here. Read [here](https://github.com/django/django/pull/14437/files#r689823015)
Thoughts on this?